enhancements/dns/coredns-egressfirewall-integration.md

```diff
@@ -0,0 +1,363 @@
+and the next lookup time based on the TTL. However, this update will only take place if there is a change in the exisiting IP
+addresses or the next lookup time or both for the DNS name.
+* The OVN-K master will watch the ``DNSName`` CRs. When the ``.status`` of a ``DNSName`` CR is updated, the OVN-K master
+will update the ``AddressSet`` for the wildcard DNS name, which is linked with the ACL rule for the corresponding EgressFirewall
+rule(s).
+* The OVN-K master will also store the regular DNS name and the corresponding current IPs along with the TTL and the next time
+to lookup. Based on the next time to lookup, the OVN-K master will follow the same method as that of the regular DNS names to get the
+latest IPs and TTL information.
+
+#### Delete/Update
+
+##### Regular DNS name
+
+* An OpenShift cluster administrator deletes an EgressFirewall resource for a Namespace containing rule(s) for regular DNS
+name(s) OR updates an EgressFirewall resource for a Namespace and deletes rule(s) containing regular DNS name(s).
+* The OVN-K master will delete the ACL rule(s) corresponding to the EgressFirewall rule(s) containing the regular DNS name(s).
+* The OVN-K master will then check if the same regular DNS names are also used in the EgressFirewall rules in other Namespaces. If
+they are not used, then the OVN-K master will delete the corrresponding ``AdressSet`` for each of the DNS names in the EgressFirewall
+rules. The OVN-K master will also delete the corresponding ``DNSName`` CRs, only if the ``.spec.isregular`` field is set to true and
+the ``.spec.iswildcard`` field is set to false. If both the fields are set to true, then the CR will not be deleted and the
+``.spec.isregular`` field will be set to false.
+
+
+##### Wildcard DNS name
+
+* An OpenShift cluster administrator deletes an EgressFirewall resource for a Namespace containing rule(s) for wildcard DNS name(s)
+OR updates an EgressFirewall resource for a Namespace and deletes rule(s) containing wildcard DNS name(s).
+* The OVN-K master will delete the ACL rule(s) corresponding to the EgressFirewall rule(s) containing the wildcard DNS name(s).
+* The OVN-K master will then check if the same wildcard DNS names are also used in the EgressFirewall rules in other Namespaces. If
+they are not used, then the OVN-K master will delete the corrresponding ``AdressSet`` for each of the DNS names in the EgressFirewall
+rules. The OVN-K master will also delete the corresponding ``DNSName`` CRs, only if the ``.spec.isregular`` field is set to false and
+the ``.spec.iswildcard`` field is set to true. If both the fields are set to true, then the CR will not be deleted and the
+``.spec.iswildcard`` field will be set to false. From the ``.status`` field all the other DNS names' details will be removed and
+only the details of the DNS name will be kept which matches with the Name field of the CR.
+
+
+#### Variation [optional]
```

Comment:
if you're not using this section you can delete it

Reply:
The EP template mentioned not to delete any section even if it's not used.